ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandStoreTest.java

```diff
@@ -292,14 +290,121 @@ public void shouldDistributeCommand() throws ExecutionException, InterruptedExce
     assertThat(record.value().getOriginalProperties(), equalTo(ksqlConfig.getAllConfigPropsWithSecretsObfuscated()));
   }
 
+  @Test
+  public void shouldIncludeTopicOffsetInSuccessfulQueuedCommandStatus()
+      throws InterruptedException, ExecutionException {
+    // Given:
+    final CommandStore commandStore = createCommandStoreThatAssignsSameId(commandId);
+
+    expect(commandProducer.send(anyObject(ProducerRecord.class))).andReturn(future);
+    expect(future.get()).andReturn(recordMetadata);
+    replay(commandProducer, future);
+
+    // When:
+    final QueuedCommandStatus commandStatus =
+        commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
+
+    // Then:
+    assertThat(commandStatus.getCommandOffset(), equalTo(recordMetadata.offset()));
+
+    verify(commandProducer, future);
+  }
+
+  @Test
+  public void shouldReturnCompletedFutureIfOffsetReached() {
+    // Given:
+    givenCmdConsumerAtPosition(1);
+
+    final CommandStore commandStore = createCommandStore();
+
+    // When:
+    final CompletableFuture<Void> future = commandStore.getConsumerPositionFuture(0);
+
+    // Then:
+    assertFutureIsCompleted(future);
+    verify(commandConsumer);
+  }
+
+  @Test
+  public void shouldReturnUncompletedFutureIfOffsetNotReached() {
+    // Given:
+    givenCmdConsumerAtPosition(2);
```

Comment:
`givenCmdConsumerAtPosition(2)` is a bit misleading. Maybe `givenCommandStoreUpToPos(1)`?

Reply:
`givenCommandStorePastPosition` may be better. `Through` seems like a strange to me.